snake_ai.py

```diff
@@ -34,13 +35,17 @@ def _best_move_helper(self, snake_id, board, depth=DEPTH):
         given board, according to the heuristic, calculated to the given depth.
         Assume other snakes make the best immediate move.
         """
-        move_evals = {
-            move: self._get_move_eval(
-                snake_id,
-                self.game.simulate_moves(board, {snake_id: move}),
-                depth)
-            for move in Move}
-        return max(move_evals.items(), key=lambda x: x[1])
+        MoveEvalPair = namedtuple('MoveEvalPair', "move, evaluation")
```

Comment:
Describing the fields of a class and their datatype is not good naming. 

EG you wouldnt call a (x , y) named tuple an XYPair. What does this _actually_ represent?

Also if you are passing around tuples, consider if it's easier to make a function of just a->b rather than a-> (a, b). I think it is OK for now but just consider this.